cli/tests/pcluster/schemas/test_cluster_schema.py

```diff
@@ -486,7 +487,33 @@ def test_scheduling_schema(mocker, config_dict, failure_message):
                     }
                 ],
             },
-            "A Compute Resource needs to specify either InstanceType or Instances.",
+            "A Compute Resource needs to specify InstanceTypes, InstanceType or CapacityReservationId.",
+        ),
+        # Failing to specify either InstanceType or Instances should return a validation error
+        (
+            {
+                "Name": "Flex-Queue",
+                "Networking": {"SubnetIds": ["subnet-12345678"]},
+                "AllocationStrategy": "lowest-price",
+                "ComputeResources": [
+                    {"Name": "compute_resource1", "CapacityReservationTarget": {"CapacityReservationId": "id"}}
+                ],
+            },
+            None,
+        ),
+        (
```

Comment:
what if both InstanceType and CapacityReservationId are specified? Or if Instances and CapacityReservationId are specified?
Can you add a test for these conditions?

Reply:
Added